modules/standard/CommDiagnostics.chpl

```diff
@@ -17,86 +17,269 @@
   // due to some shortcomings in our extern records implementation.
   // Once that gets sorted out, we can turn this into an extern record,
   // and remove the 8 or so individual functions below that return the
   // various counters.
+
+  /* Aggregated communication operation counts.  This record type is
+     defined in the same way by both the runtime and this module,
+     because we don't have a good way to inherit types back and forth
+     between the two.  This first definition duplicates the one in the
+     runtime.
+   */
   extern record chpl_commDiagnostics {
+    /*
+      blocking GETs, in which initiator waits for completion
+     */
     var get: uint(64);
+    /*
+      non-blocking GETs
+     */
     var get_nb: uint(64);
+    /*
+      blocking PUTs, in which initiator waits for completion
+     */
     var put: uint(64);
+    /*
+      non-blocking PUTs
+     */
     var put_nb: uint(64);
+    /*
+      tests for non-blocking GET/PUT completions
+     */
     var test_nb: uint(64);
+    /*
+      blocking waits for non-blocking GET/PUT completions
+     */
     var wait_nb: uint(64);
+    /*
+      non-blocking waits for non-blocking GET/PUT completions
+     */
     var try_nb: uint(64);
+    /*
+      blocking remote executions, in which initiator waits for completion
+     */
     var fork: uint(64);
+    /*
+      blocking remote executions performed by the target locale's
+      Active Message handler
+     */
     var fork_fast: uint(64);
+    /*
+      non-blocking remote executions
+     */
     var fork_nb: uint(64);
   };
-  
+
+  /*
+    The Chapel record type inherits the runtime's definition of it.
+   */
   type commDiagnostics = chpl_commDiagnostics;
-  
+
+  pragma "no doc"
   extern proc chpl_startVerboseComm();
+
+  pragma "no doc"
   extern proc chpl_stopVerboseComm();
+
+  pragma "no doc"
   extern proc chpl_startVerboseCommHere();
+
+  pragma "no doc"
   extern proc chpl_stopVerboseCommHere();
+
+  pragma "no doc"
   extern proc chpl_gen_startCommDiagnostics();
+
+  pragma "no doc"
   extern proc chpl_gen_stopCommDiagnostics();
+
+  pragma "no doc"
   extern proc chpl_gen_startCommDiagnosticsHere();
+
+  pragma "no doc"
   extern proc chpl_gen_stopCommDiagnosticsHere();
+
+  pragma "no doc"
   extern proc chpl_resetCommDiagnosticsHere();
+
+  pragma "no doc"
   extern proc chpl_getCommDiagnosticsHere(out cd: commDiagnostics);
-  
+
+  /*
+    Start real-time reporting of communication initiated on any locale.
+    The output consists of a line written to ``stdout`` for each
+    communication operation.  (Here ``stdout`` means the file associated
+    with the process, not the Chapel channel with the same name.)
+   */
   proc startVerboseComm() { chpl_startVerboseComm(); }
+
+  /*
+    Stop real-time reporting of communication initiated on any locale.
+   */
   proc stopVerboseComm() { chpl_stopVerboseComm(); }
+
+  /*
+    Start real-time reporting of communication initiated on this locale.
```

Comment:
I'd prefer to use the same wording for all of there *Here functions.  One of "initiated on this locale" or "initiated on the calling locale" works for me.